### PR TITLE
length should take an OptionalValue

### PR DESCRIPTION
**Pull request by glyn**
Fixes https://github.com/ietf-wg-jsonpath/draft-ietf-jsonpath-base/issues/383

### CODE REVIEW

File: draft-ietf-jsonpath-base.md [hunk 1]
```diff
@@ -1471,7 +1471,7 @@ according to one of the following rules:
 ### `length` Function Extension {#length}
```
Comment (cabo) [inline draft-ietf-jsonpath-base.md:1471]: The text of this section still assumes that the argument is a value.  The list of cases works out, but this is still a bit inconsistent.

Comment (glyn) [reply draft-ietf-jsonpath-base.md:1471]: Hmmm. It seemed consistent with descriptions of other functions which take optional arguments.

Comment (cabo) [reply draft-ietf-jsonpath-base.md:1471]: let's go for it for now, we can take another pass.

**Review by cabo — APPROVED** [review]

File: draft-ietf-jsonpath-base.md (hunk 1 above)
Comment (gregsdennis) [reply draft-ietf-jsonpath-base.md:1471]: > Its only argument is a value (possibly taken from a singular path as
> in the example above).  The result also is a value, an unsigned
> integer.

It uses "value" language for both input and output.  Is it safe to assume from 

> For any other argument value, the result is `Nothing`.

and typing it as `OptionalValue`, that `Nothing` is returned when `Nothing` is received?

**Comment by glyn** [comment]
@cabo I pushed cb1cf1 to clarify optional arguments.

**Review by cabo — APPROVED** [review]
That should be enough to remove this double-take.

File: draft-ietf-jsonpath-base.md (hunk 1 above)
Comment (glyn) [reply draft-ietf-jsonpath-base.md:1471]: I think that's pretty clear, yes. I've tried a clarification, but we can tweak later if necessary.

**Pull request merged**